src/spriteloader/makeindexed.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file makeindexed.cpp Implementation for converting sprites from another source from 32bpp RGBA to indexed 8bpp. */
+
+#include "../stdafx.h"
+#include "../core/bitmath_func.hpp"
+#include "../core/math_func.hpp"
+#include "../gfx_func.h"
+#include "../palette_func.h"
+#include "makeindexed.h"
+
+#include "../safeguards.h"
+
+/**
+ * Adjust brightness of colour.
+ * This is a copy of Blitter_32bppBase::ReallyAdjustBrightness.
+ * @param colour Colour to adjust.
+ * @param brightness Brightess to apply to colour.
+ * @returns Adjusted coloured.
+ */
+static Colour AdjustBrightness(Colour colour, int brightness)
+{
+	uint64_t combined = (((uint64_t) colour.r) << 32) | (((uint64_t) colour.g) << 16) | ((uint64_t) colour.b);
```

Comment:
Does this really need to be a verbatim copy? Can't we put this (as `constexpr`?) function in some header that's included and used here and in `ReallyAdjustBrightness`?
Also the casts on this line could use some love.